src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -351,9 +360,13 @@ class LedgerImpl(vm: VM, blockchain: BlockchainImpl, blockchainConfig: Blockchai
     }
   }
 
-  private def pay(address: Address, value: UInt256)(world: InMemoryWorldStateProxy): InMemoryWorldStateProxy = {
-    val account = world.getAccount(address).getOrElse(Account.empty(blockchainConfig.accountStartNonce)).increaseBalance(value)
-    world.saveAccount(address, account)
+  private[ledger] def pay(address: Address, value: UInt256)(world: InMemoryWorldStateProxy): InMemoryWorldStateProxy = {
+    if (world.isZeroValueTransferToNonExistentAccount(address, value)) {
+      world
+    } else {
+      val account = world.getAccount(address).getOrElse(Account.empty(blockchainConfig.accountStartNonce)).increaseBalance(value)
+      world.saveAccount(address, account).touchAccounts(address)
```

Comment:
hm, sorry it it's obvious but why are we touching `address` here. It can still be "normal", non-empty account, can't it?

Reply:
ok, I see. `deleteTouchedAccounts` does not delete all touched accounts, it checks for emptiness first.

Reply:
hmm i will change name to ` deleteEmptyTouchedAccounts`, to be more precise